src/main/java/org/jivesoftware/smackx/pubsub/PubSubIntegrationTest.java

```diff
@@ -915,4 +918,18 @@ public void transientNotificationOnlyNodeWithItemTest() throws NoResponseExcepti
             pubSubManagerOne.deleteNode(nodename);
         }
     }
+
+    @SmackIntegrationTest
+    public void deleteNonExistentNodeTest() throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException {
+        final String nodename = "sinttest-publish-item-nodename-" + testRunId;
+        try {
+            // Delete an unexisting node
+            pubSubManagerOne.deleteNode(nodename);
+
+            assertEquals(null, pubSubManagerOne.getNode(nodename));
+        }
+        catch (PubSubException.NotAPubSubNodeException e){
+            throw new AssertionError("The node can not be deleted",e);
+        }
+    }
```

Comment:
For an example of how to assert that an exception occurs, look at https://github.com/goodbytes/smack-sint-extensions/blob/master/src/main/java/org/jivesoftware/smackx/pubsub/PubSubIntegrationTest.java#L154-L164

In psuedo-code, you probably want something like this:

```
try 
{
  // do something that should fail.
  doSomethingThatMustFail();

  // fail the test (the code should not reach this line, as the line above should cause the exception to be thrown).
  fail("The server should have returned an error, but did not.");
} 
catch ( /* exception that you expect to happen */ )
{
  // assert that the exception is what you expect (for example: contains a specific condition).
  assert( exception has condation X );
}